plugins/outputs/mqtt/mqtt_test.go

```diff
@@ -103,10 +111,46 @@ func TestConnectAndWriteIntegrationMQTTv5(t *testing.T) {
 	}
 
 	// Verify that we can connect to the MQTT broker
-	err := m.Connect()
+	err := m.Init()
+	require.NoError(t, err)
+	err = m.Connect()
 	require.NoError(t, err)
 
 	// Verify that we can successfully write data to the mqtt broker
 	err = m.Write(testutil.MockMetrics())
 	require.NoError(t, err)
 }
+
+func TestMQTT_Init(t *testing.T) {
+	tests := []struct {
+		name    string
+		topic   string
+		wantErr bool
+	}{
+		{
+			name:    "a valid pattern is accepted",
+			topic:   "this/is/valid",
+			wantErr: false,
```

Comment:
Instead of a boolean, how about specifying the expected error message (as string)?

Reply:
I would have imagined that doing that type of comparison, couples the test strongly to the implementation and is less relevant to actual functionality I was wanting to test.
I will follow advice here, but I feel the level of coupling is less than ideal by enforcing string comparison on error messages.